sm64ex: Various Features

N00byKing opened the pull request:
This adds an option to randomize both the normal courses as well as the secret courses throughout the game.

~However, there's a problem: I keep getting fill errors for what should be totally fine seeds.
If someone could take a look that'd be great~

EDIT: Now includes logic fixes as well for Big Star Doors, Options for setting MIPS Costs

Alchav commented on worlds/sm64ex/Options.py:
```diff
@@ -44,9 +44,12 @@ class ExtraStars(Range):
+class AreaRandomizer(Choice):
+    """Randomize Entrances"""
+    display_name = "Entrance Randomizer"
+    option_Off = 0
```
need to add `alias_false = 0` so this option does not cause generation failure

Alchav reviewed (changes requested):
I generated several times and the only failure I got was related to the code comment I left. If you can get one of these fill failures to occur can you provide a yaml and seed number?

N00byKing commented:
Generation failure only occurs when using Course and Secret Randomization.
Try Seed `81022829027716158669`, [yaml attached](https://github.com/ArchipelagoMW/Archipelago/files/9130279/yaml.zip)

Alchav commented:
Appears what happened here is it placed DDD at BITFS, where the access rule requires being able to reach DDD, so DDD is unreachable.

N00byKing commented:
THANK YOU!
This was getting extremely frustrating, but you're right on the money about the issue and 380 generations later and no issues means this was hopefully the last thing.
Sometimes you really do need a fresh set of eyes to find an issue.
Fixed this, and cleaned it up a bit as well. Ready for review

espeon65536 commented:
Have now looked at and played a game with it, seems fine to merge.
